packages/@eventual/core-runtime/src/utils.ts

```diff
@@ -64,3 +69,48 @@ export function deserializeCompositeKey(
   const [name, namespace, key] = sKey.split("|") as [string, string, string];
   return [name, namespace ? { key, namespace } : key];
 }
+
+export function isEntityStreamItem(value: any): value is EntityStreamItem<any> {
+  return "entityName" in value && "operation" in value;
+}
+
+export function isBucketStreamItem(value: any): value is BucketStreamItem {
+  return "bucketName" in value && "operation" in value;
+}
+
+export function entityStreamMatchesItem(
+  item: EntityStreamItem<any>,
+  streamSpec: EntityStreamSpec
+) {
+  return (
+    streamSpec.entityName === item.entityName &&
+    (!streamSpec.options?.operations ||
+      streamSpec.options.operations.includes(item.operation)) &&
+    (!streamSpec.options?.namespaces ||
+      (item.namespace &&
+        streamSpec.options.namespaces.includes(item.namespace))) &&
+    (!streamSpec.options?.namespacePrefixes ||
+      (item.namespace &&
+        streamSpec.options.namespacePrefixes.some((p) =>
+          item.namespace?.startsWith(p)
+        )))
+  );
+}
```

Comment:
What is this doing?

Reply:
jsut moved this from core, implementation of the stream matching logic for use in code (ex: local).